internal/backup/volumesnapshotcontent_action.go

```diff
@@ -119,30 +122,104 @@ func (p *VolumeSnapshotContentBackupItemAction) Execute(item runtime.Unstructure
+			if err != nil {
+				return nil, nil, "", nil, errors.Wrapf(err, "error getting volumesnapshotbackup client")
+			}
 
 			err = vsbClient.Create(context.Background(), &vsb)
 
 			if err != nil {
-				return nil, nil, errors.Wrapf(err, "error creating volumesnapshotbackup CR")
+				return nil, nil, "", nil, errors.Wrapf(err, "error creating volumesnapshotbackup CR")
 			}
 
 			p.Log.Infof("Created volumesnapshotbackup %s", fmt.Sprintf("%s/%s", vsb.Namespace, vsb.Name))
+
+			// Now fetch the VSB so that we get the UID from VSB metadata and return that as operationID to be used for progress monitoring
+			err = vsbClient.Get(context.Background(), client.ObjectKey{Namespace: vsb.Namespace, Name: vsb.Name}, &vsb)
+			if err != nil {
+				return nil, nil, "", nil, errors.Wrapf(err, "error fetching volumesnapshotbackup CR for suppyling operationID")
+			}
+
+			// operationID for our datamover usecase is VSB NamespacedName which will unique per operation
+			operationID = vsb.Namespace + "/" + vsb.Name
+
+			p.Log.Infof("Got operationID: %s", operationID)
 		}
 
-		// adding volumesnapshotbackup instance as an additional item, need to block the plugin execution till VSB CR is recon complete
-		additionalItems = append(additionalItems, velero.ResourceIdentifier{
-			GroupResource: schema.GroupResource{Group: "datamover.oadp.openshift.io", Resource: "volumesnapshotbackup"},
+		// adding volumesnapshotbackup instance as an item that needs to be updated in backup's finalizing phase with all its annotations and status
+		itemsToUpdate = append(itemsToUpdate, velero.ResourceIdentifier{
+			GroupResource: schema.GroupResource{Group: "datamover.oadp.openshift.io", Resource: "volumesnapshotbackups"},
 			Name:          vsb.Name,
 			Namespace:     vsb.Namespace,
 		})
 	}
 
-	p.Log.Infof("Returning from VolumeSnapshotContentBackupItemAction with %d additionalItems to backup", len(additionalItems))
-	return item, additionalItems, nil
+	p.Log.Infof("Returning from VolumeSnapshotContentBackupItemActionV2 with %d itemsToUpdate to backup", len(itemsToUpdate))
+	return item, nil, operationID, itemsToUpdate, nil
+}
+
+func (p *VolumeSnapshotContentBackupItemActionV2) Progress(operationID string, backup *velerov1api.Backup) (velero.OperationProgress, error) {
+	progress := velero.OperationProgress{}
+	p.Log.Infof("Backup name in progress method is: %s", backup.Name)
+	p.Log.Infof("OperationID in progress method is: %s", operationID)
+
+	// handle empty operationID case
+	if operationID == "" {
+		return progress, biav2.InvalidOperationIDError(operationID)
+	}
+
+	// fetch the VSB matching the operationID supplied, read its status and return progress of datamovement
+	vsbClient, err := util.GetVolumeSnapshotMoverClient()
+	vsb := datamoverv1alpha1.VolumeSnapshotBackup{}
+	if err != nil {
+		return progress, errors.Wrapf(err, "error getting volumesnapshotbackup client")
+	}
+
+	splitOperationID := strings.Split(operationID, "/")
```

Comment:
Probably should check len(splitOperationID) here and error out if it's not 2 before attempting to reference the elements below.